### PR TITLE
Water Balloon Ammo/ Confusion State

### PR DESCRIPTION
**Pull request by CrazyTobi**
Created data table row and blueprint for the water balloon also updated code to incorporate the confusion state, please let me know how it is and if I missed anything, didn't add logic for the confusion figured that could be its own task but worked on adding the basics at least

### CODE REVIEW

File: Source/SteveSloth/Private/MyGenericEnemyConfusionState.cpp
Comment (Saretzky) [inline Source/SteveSloth/Private/MyGenericEnemyConfusionState.cpp:1]: Add some more unique behavior in this state, maybe wandering randomly or something

**Review by Saretzky — CHANGES_REQUESTED** [review]

File: Source/SteveSloth/Public/MyGenericEnemyConfusionState.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+class STEVESLOTH_API UMyGenericEnemyConfusionState : public UMyEnemyBaseState
+{
+	GENERATED_BODY()
+	
+private: // PRIVATE INTERNAL VARIABLES
+	AMyEnemyBaseClass* Myself;
+	USkeletalMeshComponent* MyMesh;
+
+	FVector ConfusionSpot;
+	float MaxConfusionRange;
```
Comment (Saretzky) [inline Source/SteveSloth/Public/MyGenericEnemyConfusionState.h:31]: make this a Uproperty in the enemy data

**Review by Saretzky — CHANGES_REQUESTED** [review]

**Comment by Saretzky** [comment]
I did various fixes, and I know why it's not working and how to fix it. Gonna figure out how to write what is missing

**Review by Jeff-Moreau — CHANGES_REQUESTED** [review]
Other than what I wrote looks good.. Love the Vinterp nice

**Pull request merged**